draft-wang-ppm-differential-privacy.md

```diff
@@ -255,12 +255,58 @@ with the correct noise type.
 > TODO: Chris P: we will mention Prio3SumVec because that's what we use to describe aggregator DP with amplification
 
 
-# Differential Privacy Mechanism {#dp mechanism}
+# Differential Privacy Mechanism {#dp-mechanism}
 
 This section should define a framework to apply different dp noise type to a
 client measurement and/or an aggregation.
 
-> TODO: Junye to spell out the interface of dp mechanisms, "Noise Mechanisms to Accomplish DP"
+Clients and Aggregators can implement various noise mechanisms to accomplish
+the DP guarantees described in {{budget}}. The parameters of the noise
+mechanisms should be computed based on the target DP guarantee, and can be
+passed to the initializers of the noise mechanisms.
+
+We define three methods for the noise mechanism interface `DpMechanism`:
+
+* `add_noise` that adds noise to a piece of data. This can be used by a Client
+  to add noise to its measurement.
+
+* `sample_noise` that allows `DpMechanism` to sample noise directly. This can
+  be used by an Aggregator to sample Aggregator DP noise, which then gets merged
+  with the Aggregator's aggregate share.
```

Comment:
Is this needed at this level of abstraction? IIUC sample_noise would be called by an implementation of add_noise and so seems like an implementation detail here.

Reply:
I am trying to explain why `sample_noise` exists, related to the reason that I explained here: https://github.com/wangshan/draft-wang-ppm-differential-privacy/pull/4#discussion_r1267188326